docs/src/format/table/index/vector/index.md

```diff
@@ -0,0 +1,265 @@
+# Vector Indices
+
+Lance provides a powerful and extensible secondary index system for efficient vector similarity search. 
+All vector indices are stored as regular files, making them portable and easy to manage.
+It is designed for efficient similarity search across large-scale vector datasets.
+
+## Concepts
+
+Lance splits each vector index into 3 parts - clustering, sub-index and quantization.
+
+### Clustering
+
+Lance uses Inverted File (IVF) as the primary clustering mechanism.
+IVF partitions the dataset into clusters using k-means clustering. 
+Each cluster (partition) contains vectors that are similar to the cluster centroid.
+During search, only the most relevant clusters are examined, dramatically reducing search time.
+IVF can be combined with any sub-index type and quantization method.
+
+### Sub-Index
+
+The sub-index determines how vectors are organized for search. Lance currently supports:
+
+- **FLAT**: Exact search with no approximation - scans all vectors
+- **HNSW**: Hierarchical Navigable Small World graphs for fast approximate search
+
+### Quantization
+
+The quantization method determines how vectors are stored and compressed. Lance currently supports:
+
+- **Product Quantization (PQ)**: Compresses vectors by splitting them into smaller sub-vectors and quantizing each independently
+- **Scalar Quantization (SQ)**: Applies scalar quantization to each dimension of the vector independently  
```

Comment:
Do we have rabitq here?

Reply:
Will defer to @BubbleCal on this, I also had the same comment.